@bors r-
this is not working  https://github.com/rust-lang/rust/pull/136285#issuecomment-2624263002 

````
2025-01-30T11:38:35.9453186Z 3ddd26ca0e11854152de3b22030cba1560f6ca8f
2025-01-30T11:38:35.9618451Z ##[group]Run src/ci/scripts/free-disk-space.sh
2025-01-30T11:38:35.9618881Z [36;1msrc/ci/scripts/free-disk-space.sh[0m
2025-01-30T11:38:35.9657806Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
2025-01-30T11:38:35.9658136Z env:
2025-01-30T11:38:35.9658473Z   TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
2025-01-30T11:38:35.9659380Z   TOOLSTATE_REPO_ACCESS_TOKEN: ***
2025-01-30T11:38:35.9659637Z   CI_JOB_NAME: aarch64-gnu
2025-01-30T11:38:35.9659894Z   CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
2025-01-30T11:38:35.9660202Z   HEAD_SHA: 3ddd26ca0e11854152de3b22030cba1560f6ca8f
2025-01-30T11:38:35.9660570Z   DOCKER_TOKEN: ***
2025-01-30T11:38:35.9660783Z   SCCACHE_BUCKET: rust-lang-ci-sccache2
2025-01-30T11:38:35.9661052Z   CACHE_DOMAIN: ci-caches.rust-lang.org
2025-01-30T11:38:35.9661306Z ##[endgroup]
2025-01-30T11:38:35.9756411Z ================================================================================
2025-01-30T11:38:35.9756837Z BEFORE CLEAN-UP:
2025-01-30T11:38:35.9756992Z 
2025-01-30T11:38:35.9757071Z $ df -h
2025-01-30T11:38:35.9757196Z 
2025-01-30T11:38:35.9766959Z Filesystem      Size  Used Avail Use% Mounted on
2025-01-30T11:38:35.9767330Z /dev/root        73G   27G   46G  37% /
2025-01-30T11:38:35.9767604Z tmpfs           7.8G  4.0K  7.8G   1% /dev/shm
2025-01-30T11:38:35.9767880Z tmpfs           3.2G 1000K  3.2G   1% /run
2025-01-30T11:38:35.9768166Z tmpfs           5.0M     0  5.0M   0% /run/lock
2025-01-30T11:38:35.9768481Z efivarfs        128M   16K  128M   1% /sys/firmware/efi/efivars
2025-01-30T11:38:35.9768801Z /dev/sda15       98M  6.3M   92M   7% /boot/efi
2025-01-30T11:38:35.9769093Z tmpfs           1.6G  4.0K  1.6G   1% /run/user/1001
2025-01-30T11:38:35.9771069Z ================================================================================
2025-01-30T11:38:35.9771291Z 
2025-01-30T11:38:41.8955476Z E: Unable to locate package google-chrome-stable
2025-01-30T11:38:41.8955843Z E: Unable to locate package powershell
2025-01-30T11:38:41.8956165Z E: Unable to locate package google-cloud-sdk
2025-01-30T11:38:41.8956477Z E: Unable to locate package google-cloud-cli
2025-01-30T11:38:41.8987897Z ##[error]Process completed with exit code 100.
2025-01-30T11:38:41.9201938Z Post job cleanup.
2025-01-30T11:38:41.9981899Z [command]/usr/bin/git version
2025-01-30T11:38:42.0013286Z git version 2.48.1
````

--- a/src/ci/scripts/free-disk-space.sh
+++ b/src/ci/scripts/free-disk-space.sh
@@ -1,7 +1,18 @@
 #!/bin/bash
+set -euo pipefail
 
 # Free disk space on Linux GitHub action runners
 # Script inspired by https://github.com/jlumbroso/free-disk-space
+
+isX86() {
+    local arch
+    arch=$(uname -m)
+    if [ "$arch" = "x86_64" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
 
 # print a line of the specified character
 printSeparationLine() {
@@ -14,11 +25,15 @@ printSeparationLine() {
 # compute available space
 # REF: https://unix.stackexchange.com/a/42049/60849
 # REF: https://stackoverflow.com/a/450821/408734
-getAvailableSpace() { echo $(df -a | awk 'NR > 1 {avail+=$4} END {print avail}'); }
+getAvailableSpace() {
+    df -a | awk 'NR > 1 {avail+=$4} END {print avail}'
+}
 
 # make Kb human readable (assume the input is Kb)
 # REF: https://unix.stackexchange.com/a/44087/60849
-formatByteCount() { echo $(numfmt --to=iec-i --suffix=B --padding=7 $1'000'); }
+formatByteCount() {
+    numfmt --to=iec-i --suffix=B --padding=7 "$1"'000'
+}
 
 # macro to output saved space
 printSavedSpace() {
@@ -58,11 +73,27 @@ removeDir() {
     dir=${1}
 
     local before
-    before=$(getAvailableSpace)
+    if [ ! -d "$dir" ]; then
+        echo "::warning::Directory $dir does not exist, skipping."
+    else
+        before=$(getAvailableSpace)
+        sudo rm -rf "$dir"
+        printSavedSpace "$before" "Removed $dir"
+    fi
+}
 
-    sudo rm -rf "$dir" || true
+removeUnusedDirectories() {
+    local dirs_to_remove=(
+        "/usr/local/lib/android"
+        "/usr/share/dotnet"
 
-    printSavedSpace "$before" "$dir"
+        # Haskell runtime
+        "/usr/local/.ghcup"
+    )
+
+    for dir in "${dirs_to_remove[@]}"; do
+        removeDir "$dir"
+    done
 }
 
 execAndMeasureSpaceChange() {
@@ -79,21 +110,29 @@ execAndMeasureSpaceChange() {
 # Remove large packages
 # REF: https://github.com/apache/flink/blob/master/tools/azure-pipelines/free_disk_space.sh
 cleanPackages() {
-    sudo apt-get -qq remove -y --fix-missing \
-        '^aspnetcore-.*'       \
-        '^dotnet-.*'           \
-        '^llvm-.*'             \
-        'php.*'                \
-        '^mongodb-.*'          \
-        '^mysql-.*'            \
-        'azure-cli'            \
-        'google-chrome-stable' \
-        'firefox'              \
-        'powershell'           \
-        'mono-devel'           \
-        'libgl1-mesa-dri'      \
-        'google-cloud-sdk'     \
-        'google-cloud-cli'
+    local packages=(
+        '^aspnetcore-.*'
+        '^dotnet-.*'
+        '^llvm-.*'
+        '^mongodb-.*'
+        '^mysql-.*'
+        'azure-cli'
+        'firefox'
+        'libgl1-mesa-dri'
+        'mono-devel'
+        'php.*'
+    )
+
+    if isX86; then
+        packages+=(
+            'google-chrome-stable'
+            'google-cloud-cli'
+            'google-cloud-sdk'
+            'powershell'
+        )
+    fi
+
+    sudo apt-get -qq remove -y --fix-missing "${packages[@]}"
 
     sudo apt-get autoremove -y || echo "::warning::The command [sudo apt-get autoremove -y] failed"
     sudo apt-get clean || echo "::warning::The command [sudo apt-get clean] failed failed"
@@ -101,9 +140,9 @@ cleanPackages() {
 
 # Remove Docker images
 cleanDocker() {
-    echo "Removing the following docker images:"
+    echo "=> Removing the following docker images:"
     sudo docker image ls
-    echo "Removing docker images..."
+    echo "=> Removing docker images..."
     sudo docker image prune --all --force || true
 }
 
@@ -121,16 +160,11 @@ AVAILABLE_INITIAL=$(getAvailableSpace)
 printDF "BEFORE CLEAN-UP:"
 echo ""
 
-removeDir /usr/local/lib/android
-removeDir /usr/share/dotnet
-
-# Haskell runtime
-removeDir /opt/ghc
-removeDir /usr/local/.ghcup
-
-execAndMeasureSpaceChange cleanPackages "Large misc. packages"
+execAndMeasureSpaceChange cleanPackages "Unused packages"
 execAndMeasureSpaceChange cleanDocker "Docker images"
 execAndMeasureSpaceChange cleanSwap "Swap storage"
+
+removeUnusedDirectories
 
 # Output saved space statistic
 echo ""

